src/lib/builders/tree/create.ts

```diff
@@ -0,0 +1,437 @@
+import { get, writable, type Writable } from 'svelte/store';
+import type { Defaults } from "$lib/internal/types";
+import {
+    addEventListener,
+    builder,
+    createElHelpers,
+    executeCallbacks,
+    generateId,
+    getElementByMeltId,
+    isHTMLElement,
+    kbd
+} from '$lib/internal/helpers';
+
+import type { CreateTreeViewArgs, ItemDescription, TreeParts } from './types';
+import { onMount, tick } from 'svelte';
+
+
+const defaults = {
+    collapse: false,
+} satisfies Defaults<CreateTreeViewArgs>;
+
+const ATTRS = {
+    TABINDEX: 'tabindex',
+    EXPANDED: 'aria-expanded',
+    LABELLEDBY: 'aria-labelledby',
+    DATAID: 'data-id'
+}
+
+export function createTreeViewBuilder(args: CreateTreeViewArgs) {
+    const argsWithDefaults = { ...defaults, ...args };
+    const { collapse } = argsWithDefaults;
+
+    const { name } = createElHelpers<TreeParts>('tree-view');
```

Comment:
This can be outside of the fn, no need to recreate it for every instance of a builder